pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+        dsdp_array = dsdp_re.toarray().T
+        self.dsdp = dsdp_array
+        self.dsdp = col
+        # store dsdp returned
+        dsdp_extract = []
+        # get right lines from results
+        measurement_index = []
+        measurement_names = []
+        # produce the sensitivity for fixed variables
+        zero_sens = np.zeros(len(self.param_name))
+
+        # loop over measurement variables and their time points
+        for measurement_name in self.measure.model_measure_name:
+            # get right line number in kaug results
+            if self.discretize_model is not None:
+                # for DAE model, some variables are fixed
+                try:
+                    kaug_no = col.index(measurement_name)
+                    measurement_index.append(kaug_no)
+                    # get right line of dsdp
+                    dsdp_extract.append(dsdp_array[kaug_no])
+                except:
+                    self.logger.debug('The variable is fixed:', measurement_name)
+                    # for fixed variables, the sensitivity are a zero vector
+                    dsdp_extract.append(zero_sens)
+            else:
+                kaug_no = col.index(measurement_name)
+                measurement_index.append(kaug_no)
+                # get right line of dsdp
+                dsdp_extract.append(dsdp_array[kaug_no])
+
+        # Extract and calculate sensitivity if scaled by constants or parameters.
+        # Convert sensitivity to a dictionary
+        jac = {}
+        for par in self.param_name:
+            jac[par] = []
+
+        for d in range(len(dsdp_extract)):
+            for p, par in enumerate(self.param_name):
+                # if scaled by parameter value or constant value
+                if self.scale_nominal_param_value:
+                    jac[par].append(self.param_init[par]*dsdp_extract[d][p]*self.scale_constant_value)
+                else:
+                    jac[par].append(dsdp_extract[d][p]*self.scale_constant_value)
+
+        time11 = time.time()
+        self.logger.info('Build time with direct kaug mode [s]:', time_build)
+        self.logger.info('Solve time with direct kaug mode [s]:', time_solve)
+        self.logger.info('Total wall clock time [s]:', time11-time00)
+            
+        # check if another prior experiment FIM is provided other than the user-specified one
+        if self.specified_prior is None:
+            prior_in_use = self.prior_FIM
+        else:
+            prior_in_use = self.specified_prior
+
+        # Assemble and analyze results
+        FIM_analysis = FIM_result(self.param_name,self.measure, jacobian_info=None, all_jacobian_info=jac,
+                                    prior_FIM=prior_in_use, store_FIM=self.FIM_store_name,
+                                    scale_constant_value=self.scale_constant_value)
+        
+        
+        self.jac = jac
+        FIM_analysis.build_time = time_build
+        FIM_analysis.solve_time = time_solve
```

Comment:
(For clarification, not necessarily change) Why is the jacobian stored on the DOE instance, but everything else put on the FIM_analysis object?

Reply:
Comment updated. We save this jacobian to the DOE instance for two reasons; 1) an user can have access to it without calling the result object; 2) In result object, we sometimes use part of the Jacobians to get the FIM information for part of the measurements. The Jacobian we save to DOE instance is the full Jacobian, while we can feed part of the Jacobian to the result object as needed.